### ISSUE TITLE
Feature : Option for Combined Multi and OS selection style

### ISSUE THREAD

**Issue by qtzar** [DataTables/Select#17]
When working with a long table where the end user may have to select a lot of rows you can either choose the 'os' style so that they can shift/cmd click to select large amounts of rows at once or you can choose the 'multi' style where they have to click each row individually.

The 'os' style is much faster for the end user and is the preferred option but the end users get frustrated if they mis-click without holding down a modifier key and all their previous selects disappear.

Once possible idea to resolve this end user frustration would be a combination of the 'os' and 'multi' methods where the user can use the shift/cmd modifiers as usual but clicking on a single row without the modifier keys would not cause the previous selections to become unselected.

**Comment by DataTables** [comment]
Thanks for the suggestion! Sounds like a good option for a fourth variation on the way items can be selected in the table.

The idea of the `os` style is that it behaves the same was as the Finder and Explorer in Mac and Windows so users are hopefully familiar with them. I presume Apple and Microsoft have done some research and found that works best.

As the inverse to your suggestion, I'd imagine that if the user did mean to select a single row to deselect the others (as I frequently use myself, both in the file system and DataTables) it would be quite frustrating if it didn't deselect the other rows.

I suspect one way or the other some users will be frustrated!

**Comment by chellman** [comment]
I ran into this issue as well.  The OS selection method is very useful, but for tables where checkboxes are displayed, in every OS, each checkbox when single-clicked will toggle itself, and only itself.

I have a version of the typeSelect function that adds a style I've called "shift-only" since basically I want to be able to enable the shift-select functionality and let the rest of the selections work normally.  PR on the way.
